### ISSUE TITLE
question about LstmLayer class.

### ISSUE THREAD

**Issue by nannanmath** [junhyukoh/caffe-lstm#5]
I have a question about caffe-lstm.
There are members N_ and T_ in class LstmLayer ( I found default value of N_ is 1 ) and their comments are batch size and length of sequence. I am not sure if that means: there are N_ independent sequences and each one of them has T_ frames. For example, If I have 5 videos and each one has 100 frames. I must set N_ = 5 and T_ = 100. Like this, if I have only one video with 100 frames, I must set N_ = 1 and T_ = 100.
But if I have 5 videos with different frames, say, video1 has 100 frames, video2 has 90 frames, videos3 has 96 frames, video4 has 88 frames, video5 has 99 frames, I feel N_ must be set 1 and use 'clip' to solve this training. If N_ is not 1, there is no solution.

Above is my understand about the members in LstmLayer class. Could you tell me if that is right? Thanks!

**Comment by junhyukoh** [comment]
You are understanding the role of N_ and T_ correctly.
There are one general solution and another solution for some cases:
1) Use N_=1 and clip appropriately as you said.
This is actually very inefficient because the model cannot process different streams in parallel.

2) If you are using `softmax loss` as loss function, you can put 5 videos with different frames into streams (N=5) by fixing T as the maximum length (say T=100) and filling extra input frames with zeros (x_99=0, x_100=0 for 98-length video). 
In order to remove unnecessary losses caused by extra frames, you can specify `ignore_label: -1` in the `loss_param` and provide -1 as labels for the extra time steps. 
This will set loss/gradient to 0 for the extra time steps.

I think Caffe should support a mask indicator for all loss layers so that we can selectively ignore losses for specific examples/time steps.
